diesel/src/sqlite/types/json.rs

```diff
@@ -0,0 +1,575 @@
+
+        // Set the serialized binary data to the output
+        out.set_value(buffer);
+
+        Ok(IsNull::No)
+    }
+}
+
+// Helper function to read a JSONB value from the byte stream
+fn read_jsonb_value(bytes: &[u8]) -> deserialize::Result<(serde_json::Value, usize)> {
+    if bytes.is_empty() {
+        return Err("Empty JSONB data".into());
+    }
+
+    // The first byte contains both the element type and potentially the payload size
+    let first_byte = bytes[0];
+    let element_type = first_byte & 0x0F;
+    let size_hint = (first_byte & 0xF0) >> 4;
+
+    let (payload_size, header_size) = match size_hint {
+        0x00..=0x0B => (size_hint as usize, 1), // Payload size is directly in the upper nibble
+        0x0C => {
+            if bytes.len() < 2 {
+                return Err("Invalid JSONB data: insufficient bytes for payload size".into());
+            }
+            (bytes[1] as usize, 2) // 1 additional byte for payload size
+        }
+        0x0D => {
+            if bytes.len() < 3 {
+                return Err("Invalid JSONB data: insufficient bytes for payload size".into());
+            }
+            (u16::from_be_bytes([bytes[1], bytes[2]]) as usize, 3) // 2 additional bytes
+        }
+        0x0E => {
+            if bytes.len() < 5 {
+                return Err("Invalid JSONB data: insufficient bytes for payload size".into());
+            }
+            (
+                u32::from_be_bytes([bytes[1], bytes[2], bytes[3], bytes[4]]) as usize,
+                5,
+            ) // 4 additional bytes
+        }
+        0x0F => {
+            if bytes.len() < 9 {
+                return Err("Invalid JSONB data: insufficient bytes for payload size".into());
+            }
+            (
+                usize::try_from(u64::from_be_bytes([
+                    bytes[1], bytes[2], bytes[3], bytes[4], bytes[5], bytes[6], bytes[7], bytes[8],
+                ]))
+                .map_err(Box::new)?,
+                9,
+            ) // 8 additional bytes
+        }
+        _ => return Err("Invalid payload size hint".into()),
+    };
+
+    let total_size = header_size + payload_size;
+    if bytes.len() < total_size {
+        return Err("Invalid JSONB data: insufficient bytes for value".into());
+    }
+
+    let payload_bytes = &bytes[header_size..total_size];
+
+    let value = match element_type {
+        JSONB_NULL => Ok(serde_json::Value::Null),
+        JSONB_TRUE => Ok(serde_json::Value::Bool(true)),
+        JSONB_FALSE => Ok(serde_json::Value::Bool(false)),
+        JSONB_INT => read_jsonb_int(payload_bytes, payload_size),
+        JSONB_INT5 => Err("INT5 is not supported in this implementation".into()),
+        JSONB_FLOAT => read_jsonb_float(payload_bytes, payload_size),
+        JSONB_FLOAT5 => Err("FLOAT5 is not supported in this implementation".into()),
+        JSONB_TEXT | JSONB_TEXTJ | JSONB_TEXTRAW => read_jsonb_text(payload_bytes, payload_size),
```

Comment:
I think we would need some additional handling for `JSONB_TEXT` and `JSONB_TESTJ` to handle escaped characters as as far as I remember `serde_json::Value::String` is expected to contain escaped values.

Reply:
@weiznich done fd7f5658